doc/source/text.rst

```diff
@@ -303,23 +303,23 @@ The same alignment can be used when ``others`` is a ``DataFrame``:
 Concatenating a Series and many objects into a Series
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-All one-dimensional list-likes can be combined in a list-like container (including iterators, ``dict``-views, etc.):
+Several items can be combined a list-like container (including iterators, ``dict``-views, etc.), which may contain ``Series``, ``Index``, ``PandasArray`` and ``np.ndarray``.
 
 .. ipython:: python
 
     s
     u
-    s.str.cat([u.array,
-               u.index.astype(str).array], na_rep='-')
+    s.str.cat([u, u.array, u.to_numpy()], join='left')
 
-All elements must match in length to the calling ``Series`` (or ``Index``), except those having an index if ``join`` is not None:
+All elements without an index (e.g. ``PandasArray`` and ``np.ndarray``) within the passed list-like must match in length to the calling ``Series`` (or ``Index``),
```

Comment:
`PandasArray` is *very* user-facing:
```
>>> s = pd.Series(['a', 'b' ,'c', 'd'])
>>> s.array
<PandasArray>
['a', 'b', 'c', 'd']
Length: 4, dtype: object
```

and the current example was recently changed to use `.array` instead of `.values`. I think this should be documented clearly.

Reply:
OK thanks. I have been somewhat on the sidelines for that conversation so I'll defer to @jreback specifically on this piece

Reply:
You don't need to mention ``PandasArray`` here, its not very interesting, nor relevant. Just say array-likes. and remove ``u.array``, the ``u.to_numpy()`` is the corrent idiom here.

Reply:
@jreback 
There is a big difference between `u.array` and `u.to_numpy()`:
```
>>> s = pd.Series(['a', 'b', 'c', 'd'])
>>> s.array
<PandasArray>
['a', 'b', 'c', 'd']
Length: 4, dtype: object
>>> s.to_numpy()
array(['a', 'b', 'c', 'd'], dtype=object)
```

I'm guessing `.array` will eventually replace `.values`-usage (e.g. to get rid of the index for `.str.cat`), since it is by design better suited for pandas-internal dtypes, and so the distinction above is not just an irrelevant detail IMO.

I want to show here the explicitly allowed item types to pass into a list-like, which *have* to pass:
```
nxt = others.pop(0)
[...]
if not (isinstance(nxt, (Series, Index))
        or (isinstance(nxt, np.ndarray) and nxt.ndim == 1)):
    raise ValueError(...)  # currently just a DeprecationWarning
```

Long story short, I want to show a list-like containing an `np.ndarray`, a `PandasArray` (to reiterate, this example was already changed by @TomAugspurger to use `.array` instead of `.values` in #23623), and a `Series`. (Including `Index` would be nice-to-have, but too complicated absent #22225).

Reply:
@h-vetinari ``.to_numpy()`` replaces ``.values``; ``.array`` is user-accessible, but generally is not visible to by the user. its not necessary here and is just noise.

Reply:
@jreback 
> `.to_numpy()` replaces `.values`; 

As I did in the last few commits...

> `.array` is user-accessible, but generally is not visible to by the user. its not necessary here and is just noise.

Will remove, but pinging @TomAugspurger since he added that `u.array` in this example (although likely "just" for replacing `u.values`?).